src/validations/rules/sap.sch

```diff
@@ -319,6 +319,96 @@
                 unit:override-xspec="both">SAP excluded controls are identified in the associated SSP. </sch:assert>
         </sch:rule>
     </sch:pattern>
+    
+    <sch:pattern
+        id="control-objective-selection">
+        <sch:rule
+            context="oscal:control-objective-selection">
+            <sch:let
+                name="exclude-control-objective-ids"
+                value="oscal:exclude-objective/@objective-id ! xs:string(.)" />
+            <sch:let
+                name="include-control-objective-ids"
+                value="oscal:include-objective/@objective-id ! xs:string(.)" />
+            <sch:let
+                name="matching-control-objective-ids"
+                value="$exclude-control-objective-ids[. = $include-control-objective-ids]" />
+
+            <sch:assert
+                diagnostics="include-all-or-include-objective-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAP) §4.24"
+                id="include-all-or-include-objective"
+                role="fatal"
+                test="(oscal:include-all and not(oscal:include-objective)) or (oscal:include-objective and not(oscal:include-all))">An OSCAL SAP
+                control objective selection element must have either an include-all element or include-control element(s) children.</sch:assert>
+
+            <sch:assert
+                diagnostics="duplicate-exclude-objective-and-include-objective-values-diagnostic"
+                id="duplicate-exclude-objective-and-include-objective-values"
+                role="error"
+                test="count($matching-control-objective-ids) = 0">The exclude-control and include-control sibling element @control-id values must be
+                different.</sch:assert>
+
+            <sch:assert
+                diagnostics="duplicate-include-objective-values-diagnostic"
+                id="duplicate-include-objective-values"
+                role="error"
+                test="count($include-control-objective-ids) eq count(distinct-values($include-control-objective-ids))">The
+                include-objective/@objective-id values must not be duplicated.</sch:assert>
+
+            <sch:assert
+                diagnostics="duplicate-exclude-objective-values-diagnostic"
+                id="duplicate-exclude-objective-values"
+                role="error"
+                test="count($exclude-control-objective-ids) eq count(distinct-values($exclude-control-objective-ids))">The
+                exclude-objective/@objective-id values must not be duplicated.</sch:assert>
+        </sch:rule>
+        <!-- This series of variables gets to the profile file to match the include and exclude asserts below -->
+        <sch:let
+            name="profile-href"
+            value="resolve-uri($ssp-doc/oscal:system-security-plan/oscal:import-profile/@href, base-uri())" />
```

Comment:
An SSP `import-profile` may not be importing a profile.

Reply:
Made changes to import-profile handling to check for oscal:profile element in referenced document.  At least this way the excessive errors due to missing objectives will be easier to track down for the user.